File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/net/codinsa2015/StateAlterationModelView.java
package net.codinsa2015;
import java.lang.*;
import java.util.ArrayList;
import java.io.BufferedReader;
import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.UnsupportedEncodingException;
import net.codinsa2015.StateAlterationType.*;
import net.codinsa2015.DashDirectionType.*;


@SuppressWarnings("unused")
public class StateAlterationModelView
{


	// Représente le type de l'altération d'état.
	public StateAlterationType Type;
	// Durée de base de l'altération d'état en secondes
	public Float BaseDuration;
	// Si Type contient Dash : Obtient ou définit une valeur indiquant si le dash permet traverser les
	// murs.
	public Boolean DashGoThroughWall;
	// Si Type contient Dash : type direction du dash.
	public DashDirectionType DashDirType;
	// Valeur flat dubuff / debuff (valeur positive : buff, valeur négative : debuff). La nature du
	// buff dépend de Type.
	public Float FlatValue;
	// Même que FlatValue, mais en pourcentage de dégâts d'attaque actuels de la source.
	public Float SourcePercentADValue;
	// Même que FlatValue, mais en pourcentage des HP actuels de la source.
	public Float SourcePercentHPValue;
	// Même que FlatValue, mais en pourcentage des HP max de la source.
	public Float SourcePercentMaxHPValue;
	// Même que FlatValue mais en pourcentage de l'armure actuelle de la source.
	public Float SourcePercentArmorValue;
	// Même que FlatValue, mais en pourcentage de l'AP actuelle de l'entité source.
	public Float SourcePercentAPValue;
	// Même que FlatValue mais en pourcentage de la RM actuelle de l'entité source.
	public Float SourcePercentRMValue;
	// Même que FlatValue, mais en pourcentage dedégâts d'attaque actuels del'entité de
	// destination.
	public Float DestPercentADValue;
	// Même que FlatValue, mais en pourcentage des HP actuels de l'entité de destination.
	public Float DestPercentHPValue;
	// Même que FlatValue, mais en pourcentage des HP max de l'entité de destination.
	public Float DestPercentMaxHPValue;
	// Même que FlatValue mais en pourcentage de l'armure actuelle de l'entité de destination.
	public Float DestPercentArmorValue;
	// Même que FlatValue, mais en pourcentage de l'AP actuelle de l'entité de destination.
	public Float DestPercentAPValue;
	// Même que FlatValue mais en pourcentage de la RM actuelle de l'entité de destination.
	public Float DestPercentRMValue;
	// Obtient le multiplicateur de dégâts lorsque l'altération est appliquée à une structure.
	public Float StructureBonus;
	// Obtient le multiplicateur de dégâts lorsque l'altération est appliquée sur un monstre neute.
	public Float MonsterBonus;
	// Obtient le multiplicateur de dégâts lorsque l'altération est appliquée sur un Virus.
	public Float VirusBonus;
	public StateAlterationModelView() {
	}

	public static StateAlterationModelView deserialize(BufferedReader input) throws UnsupportedEncodingException, IOException {
		StateAlterationModelView _obj =  new StateAlterationModelView();
		// Type
		StateAlterationType _obj_Type = StateAlterationType.fromValue(Integer.valueOf(input.readLine()));
		_obj.Type = _obj_Type;
		// BaseDuration
		float _obj_BaseDuration = Float.valueOf(input.readLine());
		_obj.BaseDuration = _obj_BaseDuration;
		// DashGoThroughWall
		boolean _obj_DashGoThroughWall = Integer.valueOf(input.readLine()) == 0 ? false : true;
		_obj.DashGoThroughWall = _obj_DashGoThroughWall;
		// DashDirType
		DashDirectionType _obj_DashDirType = DashDirectionType.fromValue(Integer.valueOf(input.readLine()));
		_obj.DashDirType = _obj_DashDirType;
		// FlatValue
		float _obj_FlatValue = Float.valueOf(input.readLine());
		_obj.FlatValue = _obj_FlatValue;
		// SourcePercentADValue
		float _obj_SourcePercentADValue = Float.valueOf(input.readLine());
		_obj.SourcePercentADValue = _obj_SourcePercentADValue;
		// SourcePercentHPValue
		float _obj_SourcePercentHPValue = Float.valueOf(input.readLine());
		_obj.SourcePercentHPValue = _obj_SourcePercentHPValue;
		// SourcePercentMaxHPValue
		float _obj_SourcePercentMaxHPValue = Float.valueOf(input.readLine());
		_obj.SourcePercentMaxHPValue = _obj_SourcePercentMaxHPValue;
		// SourcePercentArmorValue
		float _obj_SourcePercentArmorValue = Float.valueOf(input.readLine());
		_obj.SourcePercentArmorValue = _obj_SourcePercentArmorValue;
		// SourcePercentAPValue
		float _obj_SourcePercentAPValue = Float.valueOf(input.readLine());
		_obj.SourcePercentAPValue = _obj_SourcePercentAPValue;
		// SourcePercentRMValue
		float _obj_SourcePercentRMValue = Float.valueOf(input.readLine());
		_obj.SourcePercentRMValue = _obj_SourcePercentRMValue;
		// DestPercentADValue
		float _obj_DestPercentADValue = Float.valueOf(input.readLine());
		_obj.DestPercentADValue = _obj_DestPercentADValue;
		// DestPercentHPValue
		float _obj_DestPercentHPValue = Float.valueOf(input.readLine());
		_obj.DestPercentHPValue = _obj_DestPercentHPValue;
		// DestPercentMaxHPValue
		float _obj_DestPercentMaxHPValue = Float.valueOf(input.readLine());
		_obj.DestPercentMaxHPValue = _obj_DestPercentMaxHPValue;
		// DestPercentArmorValue
		float _obj_DestPercentArmorValue = Float.valueOf(input.readLine());
		_obj.DestPercentArmorValue = _obj_DestPercentArmorValue;
		// DestPercentAPValue
		float _obj_DestPercentAPValue = Float.valueOf(input.readLine());
		_obj.DestPercentAPValue = _obj_DestPercentAPValue;
		// DestPercentRMValue
		float _obj_DestPercentRMValue = Float.valueOf(input.readLine());
		_obj.DestPercentRMValue = _obj_DestPercentRMValue;
		// StructureBonus
		float _obj_StructureBonus = Float.valueOf(input.readLine());
		_obj.StructureBonus = _obj_StructureBonus;
		// MonsterBonus
		float _obj_MonsterBonus = Float.valueOf(input.readLine());
		_obj.MonsterBonus = _obj_MonsterBonus;
		// VirusBonus
		float _obj_VirusBonus = Float.valueOf(input.readLine());
		_obj.VirusBonus = _obj_VirusBonus;
		return _obj;
	}

	public void serialize(OutputStreamWriter output) throws UnsupportedEncodingException, IOException {
		// Type
		output.append(((Integer)(this.Type.getValue())).toString() + "\n");
		// BaseDuration
		output.append(((Float)this.BaseDuration).toString() + "\n");
		// DashGoThroughWall
		output.append((this.DashGoThroughWall ? 1 : 0) + "\n");
		// DashDirType
		output.append(((Integer)(this.DashDirType.getValue())).toString() + "\n");
		// FlatValue
		output.append(((Float)this.FlatValue).toString() + "\n");
		// SourcePercentADValue
		output.append(((Float)this.SourcePercentADValue).toString() + "\n");
		// SourcePercentHPValue
		output.append(((Float)this.SourcePercentHPValue).toString() + "\n");
		// SourcePercentMaxHPValue
		output.append(((Float)this.SourcePercentMaxHPValue).toString() + "\n");
		// SourcePercentArmorValue
		output.append(((Float)this.SourcePercentArmorValue).toString() + "\n");
		// SourcePercentAPValue
		output.append(((Float)this.SourcePercentAPValue).toString() + "\n");
		// SourcePercentRMValue
		output.append(((Float)this.SourcePercentRMValue).toString() + "\n");
		// DestPercentADValue
		output.append(((Float)this.DestPercentADValue).toString() + "\n");
		// DestPercentHPValue
		output.append(((Float)this.DestPercentHPValue).toString() + "\n");
		// DestPercentMaxHPValue
		output.append(((Float)this.DestPercentMaxHPValue).toString() + "\n");
		// DestPercentArmorValue
		output.append(((Float)this.DestPercentArmorValue).toString() + "\n");
		// DestPercentAPValue
		output.append(((Float)this.DestPercentAPValue).toString() + "\n");
		// DestPercentRMValue
		output.append(((Float)this.DestPercentRMValue).toString() + "\n");
		// StructureBonus
		output.append(((Float)this.StructureBonus).toString() + "\n");
		// MonsterBonus
		output.append(((Float)this.MonsterBonus).toString() + "\n");
		// VirusBonus
		output.append(((Float)this.VirusBonus).toString() + "\n");
	}

}

File: Codinsa2015/Codinsa2015/Server/Views/StateAlterationModelView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework;


namespace Codinsa2015.Views
{

	public class StateAlterationModelView
	{

		static Encoding BOMLESS_UTF8 = new UTF8Encoding(false);
	
		/// <summary>
		/// Représente le type de l'altération d'état.
		/// </summary>
		public StateAlterationType Type;	
		/// <summary>
		/// Durée de base de l'altération d'état en secondes
		/// </summary>
		public float BaseDuration;	
		/// <summary>
		/// Si Type contient Dash : Obtient ou définit une valeur indiquant si le dash permet traverser les
		/// murs.
		/// </summary>
		public bool DashGoThroughWall;	
		/// <summary>
		/// Si Type contient Dash : type direction du dash.
		/// </summary>
		public DashDirectionType DashDirType;	
		/// <summary>
		/// Valeur flat dubuff / debuff (valeur positive : buff, valeur négative : debuff). La nature du
		/// buff dépend de Type.
		/// </summary>
		public float FlatValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage de dégâts d'attaque actuels de la source.
		/// </summary>
		public float SourcePercentADValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage des HP actuels de la source.
		/// </summary>
		public float SourcePercentHPValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage des HP max de la source.
		/// </summary>
		public float SourcePercentMaxHPValue;	
		/// <summary>
		/// Même que FlatValue mais en pourcentage de l'armure actuelle de la source.
		/// </summary>
		public float SourcePercentArmorValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage de l'AP actuelle de l'entité source.
		/// </summary>
		public float SourcePercentAPValue;	
		/// <summary>
		/// Même que FlatValue mais en pourcentage de la RM actuelle de l'entité source.
		/// </summary>
		public float SourcePercentRMValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage dedégâts d'attaque actuels del'entité de
		/// destination.
		/// </summary>
		public float DestPercentADValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage des HP actuels de l'entité de destination.
		/// </summary>
		public float DestPercentHPValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage des HP max de l'entité de destination.
		/// </summary>
		public float DestPercentMaxHPValue;	
		/// <summary>
		/// Même que FlatValue mais en pourcentage de l'armure actuelle de l'entité de destination.
		/// </summary>
		public float DestPercentArmorValue;	
		/// <summary>
		/// Même que FlatValue, mais en pourcentage de l'AP actuelle de l'entité de destination.
		/// </summary>
		public float DestPercentAPValue;	
		/// <summary>
		/// Même que FlatValue mais en pourcentage de la RM actuelle de l'entité de destination.
		/// </summary>
		public float DestPercentRMValue;	
		/// <summary>
		/// Obtient le multiplicateur de dégâts lorsque l'altération est appliquée à une structure.
		/// </summary>
		public float StructureBonus;	
		/// <summary>
		/// Obtient le multiplicateur de dégâts lorsque l'altération est appliquée sur un monstre neute.
		/// </summary>
		public float MonsterBonus;	
		/// <summary>
		/// Obtient le multiplicateur de dégâts lorsque l'altération est appliquée sur un creep.
		/// </summary>
		public float CreepBonus;	
		public StateAlterationModelView() {
		}

		public static StateAlterationModelView Deserialize(System.IO.StreamReader input) {
			StateAlterationModelView _obj =  new StateAlterationModelView();
			// Type
			StateAlterationType _obj_Type = (StateAlterationType)Int32.Parse(input.ReadLine());
			_obj.Type = (StateAlterationType)_obj_Type;
			// BaseDuration
			float _obj_BaseDuration = Single.Parse(input.ReadLine());
			_obj.BaseDuration = (float)_obj_BaseDuration;
			// DashGoThroughWall
			bool _obj_DashGoThroughWall = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.DashGoThroughWall = (bool)_obj_DashGoThroughWall;
			// DashDirType
			DashDirectionType _obj_DashDirType = (DashDirectionType)Int32.Parse(input.ReadLine());
			_obj.DashDirType = (DashDirectionType)_obj_DashDirType;
			// FlatValue
			float _obj_FlatValue = Single.Parse(input.ReadLine());
			_obj.FlatValue = (float)_obj_FlatValue;
			// SourcePercentADValue
			float _obj_SourcePercentADValue = Single.Parse(input.ReadLine());
			_obj.SourcePercentADValue = (float)_obj_SourcePercentADValue;
			// SourcePercentHPValue
			float _obj_SourcePercentHPValue = Single.Parse(input.ReadLine());
			_obj.SourcePercentHPValue = (float)_obj_SourcePercentHPValue;
			// SourcePercentMaxHPValue
			float _obj_SourcePercentMaxHPValue = Single.Parse(input.ReadLine());
			_obj.SourcePercentMaxHPValue = (float)_obj_SourcePercentMaxHPValue;
			// SourcePercentArmorValue
			float _obj_SourcePercentArmorValue = Single.Parse(input.ReadLine());
			_obj.SourcePercentArmorValue = (float)_obj_SourcePercentArmorValue;
			// SourcePercentAPValue
			float _obj_SourcePercentAPValue = Single.Parse(input.ReadLine());
			_obj.SourcePercentAPValue = (float)_obj_SourcePercentAPValue;
			// SourcePercentRMValue
			float _obj_SourcePercentRMValue = Single.Parse(input.ReadLine());
			_obj.SourcePercentRMValue = (float)_obj_SourcePercentRMValue;
			// DestPercentADValue
			float _obj_DestPercentADValue = Single.Parse(input.ReadLine());
			_obj.DestPercentADValue = (float)_obj_DestPercentADValue;
			// DestPercentHPValue
			float _obj_DestPercentHPValue = Single.Parse(input.ReadLine());
			_obj.DestPercentHPValue = (float)_obj_DestPercentHPValue;
			// DestPercentMaxHPValue
			float _obj_DestPercentMaxHPValue = Single.Parse(input.ReadLine());
			_obj.DestPercentMaxHPValue = (float)_obj_DestPercentMaxHPValue;
			// DestPercentArmorValue
			float _obj_DestPercentArmorValue = Single.Parse(input.ReadLine());
			_obj.DestPercentArmorValue = (float)_obj_DestPercentArmorValue;
			// DestPercentAPValue
			float _obj_DestPercentAPValue = Single.Parse(input.ReadLine());
			_obj.DestPercentAPValue = (float)_obj_DestPercentAPValue;
			// DestPercentRMValue
			float _obj_DestPercentRMValue = Single.Parse(input.ReadLine());
			_obj.DestPercentRMValue = (float)_obj_DestPercentRMValue;
			// StructureBonus
			float _obj_StructureBonus = Single.Parse(input.ReadLine());
			_obj.StructureBonus = (float)_obj_StructureBonus;
			// MonsterBonus
			float _obj_MonsterBonus = Single.Parse(input.ReadLine());
			_obj.MonsterBonus = (float)_obj_MonsterBonus;
			// CreepBonus
			float _obj_CreepBonus = Single.Parse(input.ReadLine());
			_obj.CreepBonus = (float)_obj_CreepBonus;
			return _obj;
		}

		public void Serialize(System.IO.StreamWriter output) {
			// Type
			output.WriteLine(((int)this.Type).ToString());
			// BaseDuration
			output.WriteLine(((float)this.BaseDuration).ToString());
			// DashGoThroughWall
			output.WriteLine(this.DashGoThroughWall ? 1 : 0);
			// DashDirType
			output.WriteLine(((int)this.DashDirType).ToString());
			// FlatValue
			output.WriteLine(((float)this.FlatValue).ToString());
			// SourcePercentADValue
			output.WriteLine(((float)this.SourcePercentADValue).ToString());
			// SourcePercentHPValue
			output.WriteLine(((float)this.SourcePercentHPValue).ToString());
			// SourcePercentMaxHPValue
			output.WriteLine(((float)this.SourcePercentMaxHPValue).ToString());
			// SourcePercentArmorValue
			output.WriteLine(((float)this.SourcePercentArmorValue).ToString());
			// SourcePercentAPValue
			output.WriteLine(((float)this.SourcePercentAPValue).ToString());
			// SourcePercentRMValue
			output.WriteLine(((float)this.SourcePercentRMValue).ToString());
			// DestPercentADValue
			output.WriteLine(((float)this.DestPercentADValue).ToString());
			// DestPercentHPValue
			output.WriteLine(((float)this.DestPercentHPValue).ToString());
			// DestPercentMaxHPValue
			output.WriteLine(((float)this.DestPercentMaxHPValue).ToString());
			// DestPercentArmorValue
			output.WriteLine(((float)this.DestPercentArmorValue).ToString());
			// DestPercentAPValue
			output.WriteLine(((float)this.DestPercentAPValue).ToString());
			// DestPercentRMValue
			output.WriteLine(((float)this.DestPercentRMValue).ToString());
			// StructureBonus
			output.WriteLine(((float)this.StructureBonus).ToString());
			// MonsterBonus
			output.WriteLine(((float)this.MonsterBonus).ToString());
			// CreepBonus
			output.WriteLine(((float)this.CreepBonus).ToString());
		}

	}
}
